pgbouncer-cloudnative-pg.yaml

```diff
@@ -0,0 +1,53 @@
+package:
+  name: pgbouncer-cloudnative-pg
+  version: 1.22.1.15
+  epoch: 0
+  description: CloudNativePG containing PgBouncer
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - pgbouncer
+
+var-transforms:
+  - from: ${{package.version}}
+    match: '\.(\d+)$'
+    replace: '-$1'
+    to: mangled-package-version
+
+pipeline:
+  # Use fetch instead of git-checkout, if we get the repo from git we need to generate docs
+  # and the docs require pandoc which requires haskell
+  - uses: git-checkout
+    with:
+      repository: https://github.com/cloudnative-pg/pgbouncer-containers
+      expected-commit: d1bcb28762f25d48330cd2de9375cdb4c488a393
+      tag: v${{vars.mangled-package-version}}
+
+  - runs: |
+      mkdir -p "${{targets.contextdir}}"/var/log/pgbouncer
+      mkdir -p "${{targets.contextdir}}"/var/run/pgbouncer
+      mkdir -p "${{targets.contextdir}}"/etc/pgbouncer
+
+      mv /etc/pgbouncer/pgbouncer.ini "${{targets.contextdir}}"/etc/pgbouncer/pgbouncer.ini.example
+      mv /etc/pgbouncer/userlist.txt "${{targets.contextdir}}"/etc/pgbouncer/userlist.txt.example
+
+      touch "${{targets.contextdir}}"/etc/pgbouncer/pgbouncer.ini "${{targets.contextdir}}"/etc/pgbouncer/userlist.txt
+
+      # https://github.com/cloudnative-pg/pgbouncer-containers/blob/d1bcb28762f25d48330cd2de9375cdb4c488a393/Dockerfile#L78
+      # we are going to call the script from /t to be able to use it as "./entrypoint.sh" in the entrypoint of the container image
+      mkdir -p "${{targets.contextdir}}"/
+      install -Dm755 entrypoint.sh "${{targets.contextdir}}"/entrypoint.sh
```

Comment:
Would our pgbouncer package + this entrypoint script be sufficient? Unsure if we need the other stuff

Reply:
That's a good point, yep it could be sufficient unless we understand why these changes are needed: 

- https://github.com/cloudnative-pg/pgbouncer-containers/blob/main/Dockerfile#L65-L68

Reply:
After a bit of digging, I'd lean towards none of these changes or an additional image being needed. The example configs aren't used anywhere and the config their entrypoint points to is an empty file on their image. We don't need the entrypoint script and can invoke `pgbouncer` directly. Next week I'll try throwing our existing image into a deployment and see how things pan out

Reply:
I agree with you, that was my first idea but then I decided to move on with building a separate package for cloudnative-pg because of the versioning scheme is different, also there are some empty files needed and this entrypoint script but yeah none of them were quite important for the image to be working fine, the existence pgbouncer provides all atm! Thanks @EyeCantCU!

Reply:
No problem! Glad I could help. I could be wrong (on mobile GitHub), but it looks like the releases are aligned with pgbouncer: https://github.com/cloudnative-pg/pgbouncer-containers/tags

All they're (CloudNative PG) doing is building pgbouncer from source in a container image, copying the config files to locations where they'll never be parsed, and creating empty files that aren't used by anything

I've confirmed looking through the chart and pgbouncers source code that these files aren't ever needed or used. I've also validated that the entrypoint is just a direct call to pgbouncer and isn't expected by the chart